Add some unit tests on number formatting (#528)

galak75 opened the pull request:
As discussed in #528

sgillies commented on tests/test_collection.py:
```diff
@@ -349,6 +349,76 @@ def test_no_filter(self):
+            self.assertEqual(rf1['properties']['integer'], 12)
+            self.assertEqual(rf2['properties']['integer'], 12)
+
+            # floats are rounded
```
@galak75 would you be willing to make the float assertions into a separate test function? The truncation and rounding effects have different causes.

sgillies commented on tests/test_collection.py:
```diff
@@ -349,6 +349,76 @@ def test_no_filter(self):
+            self.assertEqual(rf2['properties']['integer'], 12)
+
+            # ****************************************
+            # FLOAT FORMATTING IS NOT RESPECTED...
```
Here we should also have a different test since there is a split between expected and unexpected results.

galak75 commented:
@sgillies thank you for your comments. I split assertions in separate tests. Feel free to suggest different test function names (English is not my mother tongue), or additional changes.

Also, I added some new tests, based on [your comment](https://github.com/Toblerity/Fiona/issues/528#issuecomment-355327527) : 

> The integer conversion is done by Fiona and only exists to trigger exceptions when we mistakenly try to copy non-numeric values to numeric fields

- `"12.22"` is converted to `12.22` float value before being written by the target driver
- `"invalid number"` is converted to `0`

But there is actually no exception (which might be the correct behavior)

These new tests might be not necessary if already tested else where. I can remove them if you prefer. Just let me know.

sgillies commented:
Thanks @galak75 . I'm fixing up the Windows builds and will see about merging this after AppVeyor gives us the green light.

sgillies commented on tests/test_collection.py:
```diff
@@ -392,15 +392,22 @@ def test_shape_file_properties_number_formatting(self):
+            self.assertEqual(len(c), 2)
+
+            rf1, rf2 = list(c)
+
```
Excellent !

galak75 commented:
Glad to see that ! these are only tests, but it is my first contribution to an open source project. :smiley:
